xonsh/main.py

```diff
@@ -124,6 +129,8 @@ def _pprint_displayhook(value):
 
 def premain(argv=None):
     """Setup for main xonsh entry point, returns parsed arguments."""
+    if setproctitle is not None:
+        setproctitle('xonsh')
```

Comment:
Or I guess `' '.join(['xonch'] + sys.argv[1:])` if you don't like trailing whitespace.